### PR TITLE
Client Encryption : Adds support to expose Type in EncryptionKeyWrapMetadata constructor.

### PR DESCRIPTION
**Pull request by kr-santosh**
## Description

Expose Type in EncryptionKeyWrapMetadata public constructor 

        /// <summary>
        /// Creates a new instance of key wrap metadata.
        /// </summary> 
        /// <param name="type">ProviderName of KeyStoreProvider.</param>
        /// <param name="name">Name of the metadata.</param>
        /// <param name="value">Value of the metadata.</param>
        public EncryptionKeyWrapMetadata(string type, string name, string value)

## Type of change

- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs [hunk 1]
```diff
@@ -50,6 +42,15 @@ internal EncryptionKeyWrapMetadata(string type, string name, string value, strin
             this.Algorithm = algorithm;
```
Comment (anujtoshniwal) [inline Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs:42]: Can we not remove algorithm altogether? Why keep this?

Comment (kr-santosh) [reply Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs]: fixed.

**Review by anujtoshniwal — APPROVED** [review]

**Review by abhijitpai — DISMISSED** [review]

**Review by j82w — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs [hunk 2]
```diff
@@ -25,52 +25,44 @@ private EncryptionKeyWrapMetadata()
...
+        internal string Type { get; set; }
 
         /// <summary>
         /// Serialized form of metadata.
         /// Note: This value is saved in the Cosmos DB service.
         /// Implementors of derived implementations should ensure that this does not have (private) key material or credential information.
         /// </summary>
         [JsonProperty(PropertyName = "name", NullValueHandling = NullValueHandling.Ignore)]
-        public string Name { get; private set; }
+        public string Name { get; set; }
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs:57]: Why does this need to be a public set?

Comment (kr-santosh) [reply Microsoft.Azure.Cosmos/src/Resource/ClientEncryptionKey/EncryptionKeyWrapMetadata.cs]: Yeah not required, reverted .  Making Type Public would need to access in Encryption SDK, with private setter required for JSON deserialize.

**Review by anujtoshniwal — APPROVED** [review]

**Review by abhijitpai — APPROVED** [review]

**Review by j82w — APPROVED** [review]

**Pull request merged**